tests/test_diag_coded_types.py

```diff
@@ -70,62 +70,61 @@ def test_decode_leading_length_info_type_zero_length(self) -> None:
-        self.assertEqual(byte_val, bytes([0x4, 0x34, 0x56]))
-
         dct = LeadingLengthInfoType(
             base_data_type=DataType.A_BYTEFIELD,
             bit_length=5,
             base_type_encoding=None,
             is_highlow_byte_order_raw=None,
         )
-        state = EncodeState(bytearray([]), {})
-        internal = dct.convert_internal_to_bytes(bytes([0x3]), state, bit_position=1)
-        self.assertEqual(internal, bytes([0x2, 0x3]))
+        state = EncodeState(bytearray(), parameter_values={}, cursor_bit_position=3)
+        dct.encode_into_pdu(bytes([0x3]), state)
+        self.assertEqual(state.coded_message.hex(), "0803")
 
     def test_decode_leading_length_info_type_bytefield2(self) -> None:
         dct = LeadingLengthInfoType(
             base_data_type=DataType.A_BYTEFIELD,
-            bit_length=8,
+            bit_length=5,
             base_type_encoding=None,
             is_highlow_byte_order_raw=None,
         )
-        state = EncodeState(bytearray([0x12, 0x34]), {})
-        byte_val = dct.convert_internal_to_bytes(bytes([0x0]), state, bit_position=0)
-        # Right now `bytes([0x1, 0x0])` is the encoded value.
-        # However, since bytes() is shorter and would be decoded
-        # to the same value this may be changed...
-        self.assertIn(byte_val, [b'', bytes([0x1, 0x0])])
+
+        state = EncodeState(
+            bytearray.fromhex("0000ff00"), parameter_values={}, cursor_bit_position=3)
+        dct.encode_into_pdu(bytes([0xcc]), state)
+        self.assertEqual(state.coded_message.hex(), "08ccff00")
+        self.assertEqual(state.cursor_byte_position, 2)
+        self.assertEqual(state.cursor_bit_position, 0)
 
     def test_decode_leading_length_info_type_unicode2string(self) -> None:
+        # big endian
         dct = LeadingLengthInfoType(
             base_data_type=DataType.A_UNICODE2STRING,
-            bit_length=8,
+            bit_length=7,
             base_type_encoding=None,
             is_highlow_byte_order_raw=None,
         )
-        state = DecodeState(bytes([0x12, 0x4, 0x00, 0x61, 0x00, 0x39]), cursor_byte_position=1)
+        state = DecodeState(
+            bytes([0x12, 0x8, 0x00, 0x61, 0x00, 0x39, 0xff, 0x00]),
+            cursor_byte_position=1,
+            cursor_bit_position=1)
         internal_value = dct.decode_from_pdu(state)
         self.assertEqual(internal_value, "a9")
         self.assertEqual(state.cursor_byte_position, 6)
+        self.assertEqual(state.cursor_bit_position, 0)
 
+        # little endian
         dct = LeadingLengthInfoType(
             base_data_type=DataType.A_UNICODE2STRING,
             bit_length=8,
             base_type_encoding=None,
             is_highlow_byte_order_raw=False,
         )
-        state = DecodeState(bytes([0x12, 0x4, 0x61, 0x00, 0x39, 0x00]), cursor_byte_position=1)
+        state = DecodeState(
+            bytes([0x12, 0x4, 0x61, 0x00, 0x39, 0x00, 0xff, 0x00]), cursor_byte_position=1)
```

Comment:
different bytes?

Reply:
this tests for little-endian UTF-16 encoding, so the bytes representing the two letters are byte swapped compared to what's on line 106. (the first and the trailing two bytes are just garbage to check if the code properly works with arbitrary positions.)